Userland/Utilities/nc.cpp

```diff
@@ -171,6 +178,17 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             warnln("connecting to {}:{}", inet_ntop(dst_addr.sin_family, &dst_addr.sin_addr, addr_str, sizeof(addr_str)), ntohs(dst_addr.sin_port));
         }
 
+        if (test_listening_daemons) {
+            auto connect_or_error = Core::System::connect(fd, (struct sockaddr*)&dst_addr, sizeof(dst_addr));
+            char addr_str[INET_ADDRSTRLEN];
+            if (connect_or_error.is_error()) {
+                warnln("connection to {}:{} has failed due to {}", inet_ntop(dst_addr.sin_family, &dst_addr.sin_addr, addr_str, sizeof(addr_str)), ntohs(dst_addr.sin_port), strerror(connect_or_error.error().code()));
```

Comment:
you call `inet_ntop()` on both success and error paths; maybe pull that out to before the if?

(what if `inet_ntop()` fails?)